assets/src/blocks/TopicLink/TopicLinkEditor.js

```diff
@@ -0,0 +1,228 @@
+        hide_empty: true,
+        per_page: -1,
+      }) || []
+    );
+
+    const customImage = customImageId && select('core').getMedia(customImageId);
+    const customImageFromId = customImage?.source_url;
+
+    const imageId = customImageId;
+    const imageUrl = customImageFromId;
+    const imageAlt = customImage?.alt_text;
+    const currentPostCategories = select('core/editor').getCurrentPost().categories || [];
+
+    return {
+      categoriesList,
+      imageId,
+      imageUrl,
+      imageAlt,
+      currentPostCategories,
+    };
+  }, [categoryId, customImageId, customImageFromId]);
+
+  if (!categoriesList.length) {
+    return __('Populating block\'s fields…', 'planet4-blocks-backend');
+  }
+
+  /**
+   * Sets the block's category based on available categories or current post categories.
+   */
+  const setBlockCategory = () => {
+    const postCategory = categoriesList.find(category => category.id === currentPostCategories[0]);
+    const blockCategory = categoriesList.find(category => category.id === categoryId);
+
+    let categoryData = null;
+
+    if (blockCategory) {
+      categoryData = blockCategory;
+      setAttributes({categoryId: parseInt(blockCategory.id)});
+    } else if (postCategory) {
+      categoryData = postCategory;
+      setAttributes({categoryId: parseInt(postCategory.id)});
+    } else {
+      categoryData = categoriesList[0];
+      setAttributes({categoryId: parseInt(categoriesList[0].id)});
+    }
+
+    setAttributes({categoryLink: categoryData?.link || ''});
+    setAttributes({categoryName: categoryData?.name || ''});
+  };
+
+  /**
+   * Sets the object's focal position as a CSS-compatible value.
+   * @param {Object} focalPoints - Object containing x and y focal point values.
+   * @return {string} A default focal point.
+   */
+  const setObjectPosition = focalPoints => {
+    if (!focalPoints) {
+      return '50% 50%';
+    }
+    const floatX = parseFloat(focalPoints.x).toFixed(2);
+    const floatY = parseFloat(focalPoints.y).toFixed(2);
+    setAttributes({focal_points: `${floatX * 100}% ${floatY * 100}%`});
+  };
+
+  /**
+   * Converts a focal point string into an object with x and y properties.
+   * @param {string} focalPoints - The focal points string (e.g., "50% 50%").
+   * @return {Object} An object with x and y properties as decimal values.
+   */
+  const setFocalPoint = focalPoints => {
+    const [x, y] = focalPoints.split(' ').map(value => parseFloat(value) / 100);
+    return {x, y};
+  };
+
+  setBlockCategory();
+
+  // Update attributes with image data for frontend
+  setAttributes({
+    imageUrl: imageUrl || '',
+    imageAlt: imageAlt || '',
+  });
+
+  /**
+   * Renders the block preview in the editor.
+   * @return {JSX.Element} The block preview element.
+   */
+  const renderEditInPlace = () => (
+    <section className="topic-link-block block">
+      <div className="background-image">
+        {imageUrl && (
```

Comment:
It is the expected behavior if `imageUrl` is null then render an empty `<div className="background-image">`?

Reply:
Yes, if `imageUrl` is null we still render a dark background for the block until an image is chosen.